c/meson.build

```diff
@@ -43,7 +46,8 @@ if not meson.is_subproject()
 
     test_core = executable('test_core',
         sources: ['tests/test_core.c'],
-        link_with: [tskit_lib, test_lib], c_args: extra_c_args, dependencies: kastore_dep)
+        link_with: [tskit_lib, test_lib], c_args: extra_c_args, dependencies: kastore_dep,
+        c_args: ['-DTSKIT_VERSION="@0@"'.format(meson.project_version())])
```

Comment:
I used MESON_VERSION in kastore as a way of indicating "this is the version that meson thinks we're at", since it's not used at all otherwise. I guess there's not much chance of confusion when we're just passing it in to test_core.c and we're documenting it like this? How about ``MESON_PROJECT_VERSION`` as being absolutely explicit here?